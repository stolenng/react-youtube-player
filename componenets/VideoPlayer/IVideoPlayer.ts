export default interface IVideoPlayer{
    id: string;
    onFinish: () => void;
}