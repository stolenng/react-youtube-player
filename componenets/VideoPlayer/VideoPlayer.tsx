import * as React from "react";
import YouTube from 'react-youtube';

import IVideoPlayer from './IVideoPlayer';


type State = {
    id: string
};


export default class VideoInputComponent extends React.Component<IVideoPlayer, State> {
    state: State = {
        id: ''
    };
    opts = {
        height: '450',
        width: '640',
        playerVars: {
            autoplay: 1
        }
    };
    player: YT.Player;

    constructor(props: IVideoPlayer) {
        super(props);
    }

    static getDerivedStateFromProps(nextProps, prevState): State {
        return nextProps;
    }

    onPlayerStateChange(event) {
        if (YT.PlayerState.ENDED === event.data) {
            this.props.onFinish();            
        }        
    }

    isEmpty(): string {
        return this.state.id.length > 0 ? '' : 'video-empty';
    }

    render() {
        return (
            <div className={`videos-player ${this.isEmpty()}`}>
                <YouTube
                    videoId={this.state.id}
                    opts={this.opts}
                    onStateChange={(event) => this.onPlayerStateChange(event)}
                />          
            </div>
        );
    }
}