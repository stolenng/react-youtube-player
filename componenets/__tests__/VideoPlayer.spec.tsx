import * as React from "react";
import { shallow } from "enzyme";
import * as renderer from 'react-test-renderer';

import VideoPlayer from '../VideoPlayer/VideoPlayer';

describe('VideoPlayer', () => {
    it("Init VideoPlayer Componenet", () => {
        const mockFn = jest.fn();
        
        const tree = renderer
            .create(<VideoPlayer id='Ms3nnRs1SJs' onFinish={mockFn}></VideoPlayer>)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('Test start video', () => {
        const mockFn = jest.fn();
        
        const VideoInputComponent = shallow(
            <VideoPlayer id='Ms3nnRs1SJs' onFinish={mockFn}></VideoPlayer>
        );

        const instance = VideoInputComponent.instance() as VideoPlayer;

        instance.setState({ id: 'Ms3nnRs1SJs'});
        
        //Let youtube update state
        setTimeout(() => {
            expect(instance.playerStatus).toEqual(1);                    
        }, 500);
    });

    it('Test start video', () => {
        const mockFn = jest.fn();
        
        const VideoInputComponent = shallow(
            <VideoPlayer id='Ms3nnRs1SJs' onFinish={mockFn}></VideoPlayer>
        );

        const instance = VideoInputComponent.instance() as VideoPlayer;

        instance.setState({ id: 'Ms3nnRs1SJs'});
        
        instance.setState({ id: ''});
        
        //Let youtube update state
        setTimeout(() => {
            expect(instance.playerStatus).toEqual(-1);                                
        }, 500);
    });
});