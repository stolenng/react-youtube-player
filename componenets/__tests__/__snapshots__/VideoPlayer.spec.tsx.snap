// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer Init VideoPlayer Componenet 1`] = `
<div
  className="videos-player "
>
  <span
    className=""
  >
    <div
      className={null}
      id={null}
    />
  </span>
</div>
`;
